docs/image-verification.md

```diff
@@ -0,0 +1,39 @@
+# Image Verification
+
+The following covers the default "bindir" `ImageVerifier` plugin implementation.
+
+To enable image verification, add a stanza like the following to the conainerd config:
+
+```yaml
+[plugins]
+  [plugins."io.containerd.image-verifier.v1.bindir"]
+    bin_dir = "/opt/containerd-image-verifier/bin"
+    max_verifiers = 10
+    per_verifier_timeout = "10s"
+```
+
+All files in `bin_dir`, if it exists, must be verifier executables which conform to the following API.
+
+## Image Verifier Binary API
+
+### CLI Arguments
+
+- `-name`: The given reference to the image that may be pulled.
+- `-digest`: The resolved digest of the image that may be pulled.
+
+### Image Pull Judgement
+
+Print to standard output a reason for the image pull judgement.
+
+Return an exit code of 0 to allow the image to be pulled and any other exit code to block the image from being pulled.
+
+## Image Verifier Caller Contract
+
+- If `bin_dir` does not exist or contains no files, the image verifier does not block image pulls.
+- An image is pulled only if all verifiers that are called return an "ok" judgement (exit with status code 0). In other words, image pull judgements are combined with an `AND` operator.
+- If `max_verifiers < 0`, there is no imposed limit on the number of image verifiers called.
+- If `max_verifiers >= 0` (indicating there is a limit on the number of verifiers to call) and there are more than `max_verifier` entries in `bin_dir`, only `n = max_verifiers` of the verifiers will be called and the rest will be skipped. There is no guarantee which `n` of the verifiers are called.
```

Comment:
Looking for opinions: should we lexicographically order the binaries and pick the first 10?

Reply:
This is what the current implementation does, since `os.ReadDir` sorts by filename, but I wasn't sure if we wanted to commit to that implementation. If it's an observed consistent property of the implementation, perhaps it should be documented as such, however.

Reply:
Addressed this in e146129ecb7bcdd601731dac0313db4ccee9035d